### PR TITLE
Subprocess test

### LINKED ISSUE #1179: Left over line from testing in the dark monitor
There is a line in the dark monitor left over from testing/dev:

```
# FOR TESTING
        mean_slope_dir = ...
```
This should be removed

### PR DESCRIPTION
**Pull request by york-stsci**
Testing calling the pipeline in a subprocess to see whether it will work, and whether it will speed things up by allowing the use of multiprocessing in the jump and rate steps.

### CODE REVIEW

**Comment by york-stsci** [comment]
This should now be ready for review and merge. It currently contains the following:
- Subprocess-based pipeline script that uses multi-core processing
- Update to dark monitor that removes a line of test code that was keeping it from writing outputs
- Update to bad pixel monitor that sorts through outputs appropriately and generates files without crashing

File: jwql/shared_tasks/run_pipeline.py [hunk 1]
```diff
@@ -0,0 +1,296 @@
...
+        status_f.write("{}\n".format(jump_output))
+        status_f.write("{}\n".format(pipe_output))
+        status_f.write("{}\n".format(pipe_output.replace("0_ramp", "1_ramp")))
+        status_f.write("{}\n".format(fitopt_output))
+        status_f.write("SUCCEEDED")
+    # Done.
+
+
+if __name__ == '__main__':
+    with open("/internal/data1/outputs/ops/calibrated_data/general_status.txt", "w") as status_file:
```
Comment (mfixstsci) [inline jwql/shared_tasks/run_pipeline.py]: @york-stsci is there anyway we can obscure this directory with a path from he config? Policy wise I don't think we are supposed to hard code paths/information about our servers in our code.

File: jwql/shared_tasks/run_pipeline.py [hunk 2]
```diff
@@ -0,0 +1,296 @@
...
+    parser = argparse.ArgumentParser(description='Run local calibration')
+    parser.add_argument('pipe', metavar='PIPE', type=str, help=pipe_help)
+    parser.add_argument('outputs', metavar='OUTPUTS', type=str, help=out_help)
+    parser.add_argument('working_path', metavar='PATH', type=str, help=path_help)
+    parser.add_argument('instrument', metavar='INSTRUMENT', type=str, help=ins_help)
+    parser.add_argument('input_file', metavar='FILE', type=str, help=file_help)
+    parser.add_argument('short_name', metavar='NAME', type=str, help=name_help)
+    parser.add_argument('max_cores', metavar='CORES', type=str, help=cores_help)
+
+    with open("/internal/data1/outputs/ops/calibrated_data/general_status.txt", "a+") as status_file:
```
Comment (mfixstsci) [inline jwql/shared_tasks/run_pipeline.py]: Same here about the paths.

File: jwql/shared_tasks/run_pipeline.py [hunk 3]
```diff
@@ -0,0 +1,296 @@
...
+        status_file.write("Created argument parser at {}\n".format(time.ctime()))
+
+    try:
+        args = parser.parse_args()
+    except Exception as e:
+        with open("/internal/data1/outputs/ops/calibrated_data/general_status.txt", "a+") as status_file:
```
Comment (mfixstsci) [inline jwql/shared_tasks/run_pipeline.py]: Same here about the paths.

File: jwql/shared_tasks/run_pipeline.py [hunk 4]
```diff
@@ -0,0 +1,296 @@
...
+            status_file.write("Error parsing arguments.\n")
+            status_file.write("{}".format(e))
+        raise e
+
+    with open("/internal/data1/outputs/ops/calibrated_data/general_status.txt", "a+") as status_file:
```
Comment (mfixstsci) [inline jwql/shared_tasks/run_pipeline.py]: Same here about the paths.

**Review by mfixstsci — APPROVED** [review]
@york-stsci thank you for your patience on this, I finally was able to take a look at this afternoon. Just a few house keeping items and we can get this merged in.

File: jwql/shared_tasks/run_pipeline.py (hunk 1 above)
Comment (york-stsci) [reply jwql/shared_tasks/run_pipeline.py]: Done.

File: jwql/shared_tasks/run_pipeline.py (hunk 3 above)
Comment (york-stsci) [reply jwql/shared_tasks/run_pipeline.py]: Done.

File: jwql/shared_tasks/run_pipeline.py (hunk 2 above)
Comment (york-stsci) [reply jwql/shared_tasks/run_pipeline.py]: Done

File: jwql/shared_tasks/run_pipeline.py (hunk 4 above)
Comment (york-stsci) [reply jwql/shared_tasks/run_pipeline.py]: Done

**Comment by mfixstsci** [comment]
Thank you @york-stsci, I know they are minor changes but the rest looked great, thank you for submitting the changes.

**Comment by york-stsci** [comment]
Just as a note I'm going to do a quick test on one of the servers after I update this PR from develop, just to make sure nothing odd has happened.

**Comment by mfixstsci** [comment]
@york-stsci I see you made some more changes, are these changes ready for review?

**Comment by mfixstsci** [comment]
> @york-stsci I see you made some more changes, are these changes ready for review?

Looks like a merge this morning also caused a conflict, apologies!

**Comment by york-stsci** [comment]
So, the changes are related to something that came up in testing, but had not come up in actual use – the case where you run the pipeline and tell it you want "all" outputs. The standalone script knows to treat that as "create an output file for every step", but the celery function was looking for a file named whatever_whatever_whatever_all.fits, which obviously wasn't what I was intending. Given that, as far as I know, none of the actual monitor scripts ever does that, my thought is it probably only needs a relatively short review.

**Comment by york-stsci** [comment]
I've also resolved the conflict. Fortunately it wasn't difficult. There were a couple of places where I had extra logging, and develop had changed unrelated things, but I understand how it got confused.

**Review by mfixstsci — APPROVED** [review]

**Pull request merged**